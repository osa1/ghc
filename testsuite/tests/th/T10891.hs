module Main where

import T10891_Classes

import Language.Haskell.TH

main :: IO ()
main = do
  putStrLn $(stringE . show =<< reify ''C)
  putStrLn $(stringE . show =<< reify ''C')
  putStrLn $(stringE . show =<< reify ''C'')
