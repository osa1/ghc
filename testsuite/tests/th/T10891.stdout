ClassI (ClassD [] T10891_Classes.C [KindedTV a_1627391365 StarT] [] [SigD T10891_Classes.f (ForallT [KindedTV a_1627391365 StarT] [AppT (ConT T10891_Classes.C) (VarT a_1627391365)] (AppT (AppT ArrowT (VarT a_1627391365)) (ConT GHC.Types.Int)))]) []
ClassI (ClassD [] T10891_Classes.C' [KindedTV a_1627391346 StarT] [] [OpenTypeFamilyD T10891_Classes.F [KindedTV a_1627391346 StarT] (KindSig StarT) Nothing,TySynInstD T10891_Classes.F (TySynEqn [VarT a_1627391346] (VarT a_1627391346)),SigD T10891_Classes.f' (ForallT [KindedTV a_1627391346 StarT] [AppT (ConT T10891_Classes.C') (VarT a_1627391346)] (AppT (AppT ArrowT (VarT a_1627391346)) (ConT GHC.Types.Int)))]) [InstanceD [] (AppT (ConT T10891_Classes.C') (ConT GHC.Types.Int)) []]
ClassI (ClassD [] T10891_Classes.C'' [KindedTV a_1627391345 StarT] [] [DataFamilyD T10891_Classes.Fd [KindedTV a_1627391345 StarT] (Just StarT)]) [InstanceD [] (AppT (ConT T10891_Classes.C'') (ConT GHC.Types.Int)) []]
